core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -318,6 +371,163 @@ sealed class Vec[T <: Data] private[chisel3] (gen: => T, val length: Int)
     }
     curLayer(0)
   }
+
+  /** Creates a Vec literal of this type with specified values. this must be a chisel type.
+    *
+    * @param elementInitializers literal values, specified as a pair of the Vec field to the literal value.
+    * The Vec field is specified as a function from an object of this type to the field.
+    * Fields that aren't initialized to DontCare, and assignment to a wire will overwrite any
+    * existing value with DontCare.
+    * @return a Vec literal of this type with subelement values specified
+    *
+    * Vec(2, UInt(8.W)).Lit(
+    *   1 -> 0x0A.U,
+    *   2 -> 0x0B.U
+    * )
+    * }}}
+    */
+  private[chisel3] def _makeLit(elementInitializers: (Int, Data)*): this.type = {
+
+    def checkLiteralConstruction(): Unit = {
+      val dupKeys = elementInitializers.map { x => x._1 }.groupBy(x => x).flatMap { case (k, v) =>
```

Comment:
https://stackoverflow.com/questions/24729544/how-to-find-duplicates-in-a-list
might be a cleaner way to get duplicates (without the count though - and while nice, it might not be necessary)

Reply:
This was done, used maxBy